adapters/sinatra/lib/opentelemetry/adapters/sinatra/extensions/tracer_extension.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative '../middlewares/tracer_middleware'
+
+module OpenTelemetry
+  module Adapters
+    module Sinatra
+      module Extensions
+        module TracerExtension
+          # Sinatra hook after extension is registered
+          def self.registered(app)
+            # Create tracing `render` method
+            ::Sinatra::Base.module_eval do
+              def render(_engine, data, *)
+                Sinatra::Adapter.tracer.in_span(
+                  'sinatra.render_template'
+                ) do |span|
+                  template_name = data.is_a?(Symbol) ? data : :literal
+                  span.set_attribute('sinatra.template_name', template_name.to_s)
```

Comment:
`template_name` can be computed before the `span` is created. It will be more efficient to provide the attribute to `in_span`. I.e.:
```ruby
def render(_engine, data, *)
  template_name = data.is_a?(Symbol) ? data : :literal
  Sinatra::Adapter.tracer.in_span(
    'sinatra.render_template',
    attributes: {'sinatra.template_name' => template_name.to_s}
  ) do
    super
  end
end
```

Reply:
Updated to pass data when span is created (for performance). :heavy_check_mark: